src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -331,8 +331,8 @@ protected function instantiateObject(array &$data, string $class, array &$contex
             $params = [];
             foreach ($constructorParameters as $constructorParameter) {
                 $paramName = $constructorParameter->name;
+                $attributeContext = $this->getAttributeDenormalizationContext($class, $paramName, $context);
```

Comment:
here `getAttributeDenormalizationContext()` was getting passed `$key` which represents the serializedName, although we want the name of the property, to access its metadata